File: 1548-check-if-all-1s-are-at-least-length-k-places-away/check-if-all-1s-are-at-least-length-k-places-away.kt
class Solution {
    fun kLengthApart(nums: IntArray, k: Int): Boolean {
        val n: Int = nums.size

        var prev: Int = -100000
        for (i in 0 .. n-1) {
            if (nums[i] == 1) {
                if (i - prev - 1 < k) {
                    return false
                }
                prev = i
            }
        }

        return true;
    }
}

File: 1548-check-if-all-1s-are-at-least-length-k-places-away/check-if-all-1s-are-at-least-length-k-places-away.cpp
class Solution {
public:
    bool kLengthApart(vector<int>& nums, int k) {
        int n = nums.size();

        for (int i = 0, prev = -10000; i < n; i++) {
            if (nums[i] == 1) {
                if (i - prev - 1 < k) {
                    return false;
                }
                prev = i;
            }
        }

        return true;
    }
};